### PR TITLE
Fix `CloudRequestEngine` raising errors outside `Pipeline(...).process()` and stop making requests on its initialization

### LINKED ISSUE #16: CloudRequestEngine should not throw on construction if cloud.51degrees.com is unavailable
# See other
* dotnet issue https://github.com/51Degrees/pipeline-dotnet/issues/44
* java issue https://github.com/51Degrees/pipeline-java/issues/24
* php issue https://github.com/51Degrees/pipeline-php-cloudrequestengine/issues/27
# Background

If cloud.51degrees.com happens to be unavailable - the Pipeline that contains CloudRequestEngine will fail to construct, because CloudRequestEngine will try to eagerly load [these properties](https://github.com/51Degrees/pipeline-python/blob/main/fiftyone_pipeline_cloudrequestengine/fiftyone_pipeline_cloudrequestengine/cloudrequestengine.py#L98C1-L100C54) on initialization. 

# Solution
The design decision is that `CloudProperties` and `CloudEvidenceKeys` should be converted into lazy-loaded properties and should be requested from the server (requested once and then stored as instance variables) - at the point of their first use during `Process`ing of the evidence. 

This will allow any exceptions thrown from their initialization to be controlled by the `SuppressProcessExceptions` option value and if it is set to `true` - the exceptions won't propagate up and fail the whole request to the server.  This should prevent the downtimes of the main service due to downtime of `cloud.51degrees.com`.

The updated design is described in the [spec](https://github.com/51Degrees/specifications/blob/main/pipeline-specification/pipeline-elements/cloud-request-engine.md#updated-design).

# Implementation details
1. First need to reproduce these crashes using the provided example (and simulating unavailability of cloud.51degrees.com via f.e. /etc/hosts or using some proxy to intercept requests).
2. Then lazify loading these properties
3. Any exceptions thrown should now happen within Pipeline.Process() - verify that if `SuppressProcessExceptions` flag is set to true in configuration - the exceptions will actually be suppressed and added as errors into `flowData.errors`. 

# Test coverage
Current tests cover the happy path, so we need to test the unhappy paths with and without `SuppressProcessExceptions`.  In case  `SuppressProcessExceptions` is set to `True` - no exceptions are thrown during Process(), but they are returned as flowData.errors.  Otherwise we expect the exceptions be thrown but exactly from Process() method - for these properties won't be able to load.

### PR DESCRIPTION
**Pull request by BohdanVV**
Among the way, method `validate_response` of `CloudRequestEngine` was refactored due to poor quality 

Closes #16

### CODE REVIEW

File: fiftyone_pipeline_core/fiftyone_pipeline_core/flowdata.py [hunk 1]
```diff
@@ -65,24 +64,25 @@ def process(self):
         """
 
         if not self.processed:
+            # filter evidence before processing
+            try:
+                self.evidence.filter()
```
Comment (justadreamer) [inline fiftyone_pipeline_core/fiftyone_pipeline_core/flowdata.py:69]: please check if there are already tests verifying that evidence filtering works.  if not - please add them.  everything else looks good

Comment (BohdanVV) [reply fiftyone_pipeline_core/fiftyone_pipeline_core/flowdata.py:69]: It works, at least these tests are passing

https://github.com/51Degrees/pipeline-python/blob/9fcc7372238e521c77e7f1d7598a34439675498f/fiftyone_pipeline_core/tests/test_core.py#L47-L59

**Comment by BohdanVV** [comment]
I've also refactored other tests for `fiftyone_cloudrequestengine`, since they were poorly written

Should be merged now @justadreamer

**Review by justadreamer — APPROVED** [review]

**Comment by BohdanVV** [comment]
This PR was extended with the following changes:

- Resolved an issue associated with a deprecated feature in `setuptools >= 69`
- Stripped whitespace characters from the version string coming from `version.txt` to eliminate warnings
- Moved package code to the `src` subfolder to mitigate import conflicts
- Set `pytest` to use the default  `--import-mode=prepend` mode
- Addressed and fixed issues with coverage reports

**Pull request merged**